docs/src/basics/structs.md

```diff
@@ -0,0 +1,37 @@
+# Immutable `struct` Handling
+
+Problem-related types in DifferentialEquations.jl are immutable.  This
+helps, e.g., parallel solvers to efficiently handle problem types.
+
+However, you may want to modify the problem after it is created.  For
+example, to simulate it for longer timespan.  It can be done by the
+`remake` function:
+
+```julia
+prob1 = ODEProblem((u,p,t) -> u/2, 1.0, (0.0,1.0))
+prob2 = remake(prob1; tspan=(0.0,2.0))
+```
+
+A general syntax of `remake` is
+
+```julia
+modified_struct = remake(original_struct;
+  field_1 = value_1,
+  field_2 = value_2,
+  ...
+)
+```
+
+where `field_N` and `value_N` are renamed to appropriate field names
+and new desired values.
+
+Note that `remake` is not restricted to problem types.  For example,
+`SplitFunction` used via `SplitODEProblem` can be modified by `remake`
```

Comment:
`SplitFunction` isn't documented anywhere else. I am wondering if I should just be considering it an internal implementation detail.

Reply:
I think it's better to expose less when it comes to API.  So let's remove it from the first version of the document.